---
output: github_document
---

<!-- README.md is generated from README.Rmd. Please edit that file -->

```{r, include = FALSE}
knitr::opts_chunk$set(
  collapse = TRUE,
  comment = "#>",
  fig.path = "man/figures/README-",
  out.width = "100%"
)
```

# lglasso

<!-- badges: start -->

[![CRAN status](https://www.r-pkg.org/badges/version/lglasso)](https://CRAN.R-project.org/package=lglasso)
<!-- badges: end -->

<div style="text-align: justify"> 

The previous version implementes the algorithms proposed in *Zhou et al. (2024)*,  which aims to estimate the high-dimensional networks  from longitudinal data using Gassian grapical models. Though the overarching goal of the package is the same, i.e., explore the possible associations between high-dimensional data to  improve the estimations of networks. this updated version add three important  functionalities, which are

(1) Estimation of heterogeneous networks for longitudinal data. The models in previous version (i.e., in the 2024 paper) assumed a stationary process for the longitudinal data. This might not be the case in many situations, e.g, the antibody network before and after vaccination,  metabolite network before and after the initialization of the treatment for cancer patients.  In this version of *lglasso*, the function *lglasso* are extended to accommodate such important scenarios.   

(2) Extension to general clustered data. This extension is motivated by our study of metabolome data in different tissues of mice. We have 
the metabolome data in colon, ileum,  portal blood peripheral blood from same mouse. It has been observed that these data are closely correlated which is not a surprise since they are from same mouse.  When it comes to interaction networks, this package extends the algorithms for longitudinal data to such general clustered data and can estimate both general and tissue-wise networks.   


(3) This version provide function  *CVlglasso* to facilitate the selection of tuning parameter. 


</div>
## Installation

You can install the development version of lglasso from [GitHub](https://github.com/) with:

First, install the package remotes:

```
install.packages("remotes")
```

Then install lglasso :

```
remotes::install_github("jiezhou-2/lglasso", ref ="main") 
```



## How to use

 Please see [package website](https://jiezhou-2.github.io/lglasso/). 
 
 
**Reference** 

[1] Zhou J, Gui J, Viles WD, Chen H, Li S, Madan JC, Coker MO, Hoen AG. Identifying stationary microbial interaction networks based on irregularly spaced longitudinal 16S rRNA gene sequencing data. Front Microbiomes. 2024;3:1366948. doi: 10.3389/frmbi.2024.1366948. Epub 2024 Jun 2. PMID: 40687607; PMCID: PMC12276884.

[2] Friedman J., Hastie T., Tibshirani R. (2019)  Graphical Lasso: Estimation of Gaussian Graphical Models, Version: 1.11.


[3] Matt Galloway (2025),  CVglasso: Lasso Penalized Precision Matrix Estimation, version 1.0

[4] Danaher P, Wang P, Witten DM. The joint graphical lasso for inverse covariance estimation across multiple classes. J R Stat Soc Series B Stat Methodol. 2014 Mar;76(2):373-397. doi: 10.1111/rssb.12033. PMID: 24817823; PMCID: PMC4012833.
