---
title: "Identify networks from correlated data (2)"
output: rmarkdown::html_vignette
vignette: >
  %\VignetteIndexEntry{Introduction_of_lglasso_package}
  %\VignetteEngine{knitr::rmarkdown}
  %\VignetteEncoding{UTF-8}
---

```{r, include = FALSE}
knitr::opts_chunk$set(
  collapse = TRUE,
  comment = "#>"
)
```







```{r setup, message=FALSE,warning=FALSE, echo=FALSE}
library(devtools)
library(forcats)
library(fake)
library(CVglasso)
load_all()
source("../scripts/simulations.R")
set.seed(3)
```



```{r, message=FALSE}
# number of nodes
p=20
# number of edge in general network
m1=100
# the difference between the number of edges in individual networks and general network
m2=0
#  number of clusters
m3=4
# number of subjects
n=20
# correlation between cluster
rho=0.8

```


## Model 1:  homogeneous individuals

```{r}
set.seed(5)
dd=Simulate(type="longihomo",n=n,p=p,m1=m1,tt=10,tau=c(2,1))
data1=cbind(1,dd$data$pre)
colnames(data1)[1]="group"
data2=cbind(2,dd$data$post)
colnames(data2)[1]="group"
ddata=rbind(data1,data2)


lambda1=exp(seq(-5,-1,length=5))
lambda2=lambda1[1:2]
lambda=expand.grid(lambda1,lambda2)
```


### one-stage model

```{r}
bb=CVlglasso(data=data1[,-1],lambda = lambda1,K=3,cores=14,trace = FALSE)
plot(bb)
```


### two-stage model

```{r}
bb=CVlglasso(data=ddata[,-1],lambda = lambda,group=ddata[,1],K=3,cores=14,trace = FALSE)
plot(bb)
```


## Model 2:  heterogeneous individuals

```{r}
set.seed(5)
dd=Simulate(type="longiheter",n=n,p=p,m1=m1,alpha=2,group=2)
data1=cbind(1,dd$data$pre)
colnames(data1)[1]="group"
data2=cbind(2,dd$data$post)
colnames(data2)[1]="group"
ddata=rbind(data1,data2)

lambda1=exp(seq(-5,0,length=5))
lambda2=exp(seq(-5,-0,length=2))
lambda=expand.grid(lambda1,lambda2)

```


### one-stage model

```{r}

bb=CVlglasso(data=data1[,-1],lambda = lambda1,K=3,cores=14,random=TRUE)
plot(bb)
```


### two-stage model

```{r}
bb=CVlglasso(data=ddata[,-1],group=ddata[,1],lambda = lambda,K=3,cores=14,random = TRUE)
plot(bb)
```



[1] Zhou J, Gui J, Viles WD, Chen H, Li S, Madan JC, Coker MO, Hoen AG. Identifying stationary microbial interaction networks based on irregularly spaced longitudinal 16S rRNA gene sequencing data. Front Microbiomes. 2024;3:1366948. doi: 10.3389/frmbi.2024.1366948. Epub 2024 Jun 2. PMID: 40687607; PMCID: PMC12276884.

[2] Matt Galloway (2025),  CVglasso: Lasso Penalized Precision Matrix Estimation, version 1.0 

