---
title: " Identify networks from longitudinal data: homogeneous vs heterogeneous model"
output: rmarkdown::html_vignette
vignette: >
  %\VignetteIndexEntry{Introduction_of_lglasso_package}
  %\VignetteEngine{knitr::rmarkdown}
  %\VignetteEncoding{UTF-8}
---

```{r, include = FALSE}
knitr::opts_chunk$set(
  collapse = TRUE,
  comment = "#>"
)
```







```{r setup, message=FALSE,warning=FALSE,echo=FALSE}
#library(lglasso)
library(devtools)
library(forcats)
library(fake)
library(igraph)
library(Matrix)
load_all()
source("../scripts/simulations.R")
```



```{r, message=FALSE}
# number of nodes
p=10
# number of edge in general network
m1=20
# the difference between the number of edges in individual networks and general network
m2=2
# number of subjects
n=20


```



## One-stage model

### Estimate the network based on homogeneous one-stage model

#### Generate data 


```{r,warning=FALSE}

set.seed(1)
dd=Simulate(type="longihomo",n=n,p=p,m1=m1,m2=m2,tau=2,tt=10)
ddata=dd$data
subjects=fct_inorder(ddata$subject)
group=rep(0,nrow(ddata))
homoData=list(data=ddata)
saveRDS(homoData,file="homoOneStageData.rds")
```



#### Estimation

```{r,warning=FALSE}
set.seed(1)
aa=lglasso(data=ddata,lambda = 0.01,trace=T)
saveRDS(aa,file="homoOneStageResult.rds")
## estimated network 
estimates=lapply(aa$wi,function(ll){ifelse(abs(ll)>10^(-3),1,0)})
estimates
## true network 
dd$network

## correlation parameter
aa$tau
```


### Estimate the network based on heterogeneous one-stage model

#### Generate data 

```{r,warning=FALSE}
set.seed(1)
dd=Simulate(type="longiheter",n=n,p=p,m1=m1,m2=m2,tt=10,alpha=5,group=1)
ddata=dd$data$pre
subjects=fct_inorder(ddata$subject)
heterData=list(data=ddata)
saveRDS(heterData,file="heterOneStageData.rds")
```

#### Estimation

```{r,warning=FALSE}
set.seed(1)
aa=lglasso(data=ddata,lambda = 0.01,random=TRUE,trace=T)
saveRDS(aa,file="heterOneStageResult.rds")
## estimated network  

estimates=lapply(aa$wi,function(ll){ifelse(abs(ll)>10^(-3),1,0)})
estimates
##  true network
dd$network

## correlation parameter
 model <- lm(dd$tau ~ aa$tau)
plot(aa$tau,dd$tau,xlab = "estimated tau",ylab="true tau")
abline(model, col = "red", lwd = 2)
summary(model)
```


## Two-stage model


### Estimate the networks based on homogeneous two-stage model


#### Generate data 

```{r,warning=FALSE}
set.seed(1)
dd=Simulate(type="longihomo",n=n,p=p,m1=m1,m2=m2,tau=c(2,1),tt=2)
ddata=do.call(rbind,dd$data)
subjects=fct_inorder(ddata$subject)
group=c(rep(0,nrow(ddata)/2),rep(1,nrow(ddata)/2))
homoTwoStageData=list(data=ddata,group=group)
saveRDS(homoTwoStageData,file="homoTwoStageData.rds")
```

#### Estimation

```{r,warning=FALSE}
set.seed(1)
aa=lglasso(data=ddata,lambda = c(0.1,0.1),group = group,trace=T)
saveRDS(aa,file="homoTwoStageResult.rds")
estimates=lapply(aa$wi,function(ll){ifelse(abs(ll)>10^(-3),1,0)})

## pre-treatment network estimate 
estimates[[1]]

##  true post-treatment network
dd$network$pre


## post-treatment network estimate 
estimates[[2]]

##  true post-treatment network
dd$network$post
## correlation parameters
aa$tau
```


### Estimate the networks based on hetergeneous two-stage model


#### Generate data 

```{r,warning=FALSE}
set.seed(1)
dd=Simulate(type="longiheter",n=n,p=p,m1=m1,m2=m2,tt=20,alpha=0.5,group=2)
ddata=do.call(rbind,dd$data)
subjects=fct_inorder(ddata$subject)
group=c(rep(0,nrow(ddata)/2),rep(1,nrow(ddata)/2))
heterTwoStageData=list(data=ddata,group=group)
saveRDS(heterTwoStageData,file="heterTwoStageData.rds")
```


#### Estimation

```{r,warning=FALSE}
set.seed(1)
aa=lglasso(data=ddata,lambda = c(0.1,0.1),random=T,group = group,trace=T)
saveRDS(aa,file="heterTwoStageResult.rds")
estimates=lapply(aa$wi,function(ll){ifelse(abs(ll)>10^(-3),1,0)})

## pre-treatment network estimate 
estimates[[1]]

##  true pre-treatment network
dd$network$pre


## post-treatment network estimate 
estimates[[2]]

##  true post-treatment network
dd$network$post

## correlation parameters
 model <- lm(dd$tau ~ aa$tau)
plot(aa$tau,dd$tau,xlab = "estimated tau",ylab="true tau")
abline(model, col = "red", lwd = 2)
summary(model)$coef
```


[1] Zhou J, Gui J, Viles WD, Chen H, Li S, Madan JC, Coker MO, Hoen AG. Identifying stationary microbial interaction networks based on irregularly spaced longitudinal 16S rRNA gene sequencing data. Front Microbiomes. 2024;3:1366948. doi: 10.3389/frmbi.2024.1366948. Epub 2024 Jun 2. PMID: 40687607; PMCID: PMC12276884.


[2] Friedman J., Hastie T., Tibshirani R. (2019)  Graphical Lasso: Estimation of Gaussian Graphical Models, Version: 1.11. https://CRAN.R-project.org/package=glasso. 

