---
title: " Identify networks from correlated data (1)"
output: rmarkdown::html_vignette
vignette: >
  %\VignetteIndexEntry{Introduction_of_lglasso_package}
  %\VignetteEngine{knitr::rmarkdown}
  %\VignetteEncoding{UTF-8}
---

```{r, include = FALSE}
knitr::opts_chunk$set(
  collapse = TRUE,
  comment = "#>"
)
```







```{r setup, message=FALSE,warning=FALSE}
#library(lglasso)
library(devtools)
load_all()
```



```{r, message=FALSE}
# number of nodes
p=5
# number of edge in general network
m1=5
# the difference between the number of edges in individual networks and general network
m2=0
#  number of tissues
m3=4
# number of subjects
n=20
# correlation between tissues
rho=0.8

```

## Model 1: clustered data with no correlation structure
```{r, message=FALSE}
set.seed(5)
cc=matrix(rho,ncol = m3,nrow = m3)
diag(cc)=1
dd=Simulate(type="general",n=n,p=p,m1=m1,m2=m2,m3=m3,cc=cc)
ddata=dd$data
```



### estimate general networks

```{r}
aa=lglasso(data=ddata,lambda = 0.01,type="general",tol = 0.01, trace=TRUE)

## precision matrix
aa$wi

## correlation between different tissues
cov2cor(aa$v[[1]])
```

### estimate individual networks
```{r}
group=ifelse(ddata[,2]%in%c("tissue1","tissue2"),1,2)
aa=lglasso(data=ddata,lambda = c(0.001,0.001),type="general",group = group, ,tol = 0.01, trace=TRUE)
aa$wi
cov2cor(aa$v[[1]])
```

## Model 2: longitudinal data with one-parameter structured correlation 


```{r,eval=F}
library(forcats)
set.seed(5)
dd=Simulate(type="longihomo",n=n,p=p,m1=m1,tau=c(2,1))
ddata=dd$data
subjects=fct_inorder(ddata$subject)
nn=table(subjects)
group=c()
for (i in 1:length(nn)) {
 if (nn[i]>2){
   group=c(group,c(1,1,rep(2,nn[i]-2)))
 }else{
   group=c(group,rep(1,nn[i]))
 } 
}
```





### estimate general networks 

```{r,eval=F}
aa=lglasso(data=ddata,lambda = 0.1,type="expFixed",expFix=1,tol = 0.01, trace=TRUE)
aa$wi
```


### estimate individual networks

```{r,eval=FALSE}
aa=lglasso(data=ddata,lambda = c(0.1,0.1),type="expFixed",expFix=1,group = group, tol = 0.01, trace=TRUE)
aa$wiList
```





[1] Zhou J, Gui J, Viles WD, Chen H, Li S, Madan JC, Coker MO, Hoen AG. Identifying stationary microbial interaction networks based on irregularly spaced longitudinal 16S rRNA gene sequencing data. Front Microbiomes. 2024;3:1366948. doi: 10.3389/frmbi.2024.1366948. Epub 2024 Jun 2. PMID: 40687607; PMCID: PMC12276884.


[2] Friedman J., Hastie T., Tibshirani R. (2019)  Graphical Lasso: Estimation of Gaussian Graphical Models, Version: 1.11. https://CRAN.R-project.org/package=glasso. 

