---
title: " Identify networks from longitudinal data: homogeneous vs heterogeneous model"
output: rmarkdown::html_vignette
vignette: >
  %\VignetteIndexEntry{Introduction_of_lglasso_package}
  %\VignetteEngine{knitr::rmarkdown}
  %\VignetteEncoding{UTF-8}
---

```{r, include = FALSE}
knitr::opts_chunk$set(
  collapse = TRUE,
  comment = "#>"
)
```







```{r setup, message=FALSE,warning=FALSE,echo=FALSE}
#library(lglasso)
library(devtools)
library(forcats)
load_all()
set.seed(2)
```



```{r, message=FALSE}
# number of nodes
p=20
# number of edge in general network
m1=20
# the difference between the number of edges in individual networks and general network
m2=0
# number of subjects
n=20


```



## One-stage model



### Estimate the network based on homogeneous one-stage model


#### Generate data 


```{r,warning=FALSE}


dd=Simulate(type="longihomo",n=n,p=p,m1=m1,m2=m2,tau=2,tt=10)
ddata=dd$data
subjects=fct_inorder(ddata$subject)
group=rep(0,nrow(ddata))
```



#### Estimation

```{r,warning=FALSE}
aa=lglasso(data=ddata,lambda = 0.01,type="expFixed",expFix=1,tol = 0.01, trace=TRUE)

## estimated vs true network
estimates=lapply(aa$wi,function(ll){ifelse(abs(ll)>10^(-3),1,0)})
estimates
dd$network

## correlation parameter
aa$tau
```


### Estimate the network based on heterogeneous one-stage model

#### Generate data 

```{r,warning=FALSE}
dd=Simulate(type="longiheter",n=n,p=p,m1=m1,m2=m2,tt=10,alpha=5,group=1)
ddata=dd$data$pre
subjects=fct_inorder(ddata$subject)
```

#### Estimation

```{r,warning=FALSE}
aa=lglasso(data=ddata,lambda = 0.01,type="expFixed",random=TRUE,expFix=1,tol = 0.01, trace=TRUE,maxit = 30,N=100)

## estimated vs true network

estimates=lapply(aa$wi,function(ll){ifelse(abs(ll)>10^(-3),1,0)})
estimates
dd$network

## correlation parameter
 model <- lm(dd$tau ~ aa$tau)
plot(aa$tau,dd$tau,xlab = "estimated tau",ylab="true tau")
abline(model, col = "red", lwd = 2)
summary(model)$coef
```


## Two-stage model


### Estimate the networks based on homogeneous two-stage model


#### Generate data 

```{r,warning=FALSE}
dd=Simulate(type="longihomo",n=n,p=p,m1=m1,m2=m2,tau=c(2,1),tt=5)
ddata=do.call(rbind,dd$data)
subjects=fct_inorder(ddata$subject)
group=c(rep(0,nrow(ddata)/2),rep(1,nrow(ddata)/2))
```

#### Estimation

```{r,warning=FALSE}
aa=lglasso(data=ddata,lambda = c(0.1,0.1),type="expFixed",expFix=1,group = group, tol = 0.01, trace=TRUE)
estimates=lapply(aa$wi,function(ll){ifelse(abs(ll)>10^(-3),1,0)})

## pre-treatment network estimate vs true network
estimates[[1]]
dd$network$pre


## post-treatment network estimate vs true network
estimates[[2]]
dd$network$post
## correlation parameters
aa$tau
```


### Estimate the networks based on hetergeneous two-stage model


#### Generate data 

```{r,warning=FALSE}
dd=Simulate(type="longiheter",n=n,p=p,m1=m1,m2=m2,tt=5,alpha=5,group=2)
ddata=do.call(rbind,dd$data)
subjects=fct_inorder(ddata$subject)
group=c(rep(0,nrow(ddata)/2),rep(1,nrow(ddata)/2))
```


#### Estimation

```{r,warning=FALSE}

aa=lglasso(data=ddata,lambda = c(0.1,0.1),type="expFixed",random=T,expFix=1,N=100,group = group, tol = 0.01,maxit = 30, trace=TRUE)

estimates=lapply(aa$wi,function(ll){ifelse(abs(ll)>10^(-3),1,0)})

## pre-treatment network estimate vs true network
estimates[[1]]
dd$network$pre


## post-treatment network estimate vs true network
estimates[[2]]
dd$network$post

## correlation parameters
 model <- lm(dd$tau ~ aa$tau)
plot(aa$tau,dd$tau,xlab = "estimated tau",ylab="true tau")
abline(model, col = "red", lwd = 2)
summary(model)$coef
```


[1] Zhou J, Gui J, Viles WD, Chen H, Li S, Madan JC, Coker MO, Hoen AG. Identifying stationary microbial interaction networks based on irregularly spaced longitudinal 16S rRNA gene sequencing data. Front Microbiomes. 2024;3:1366948. doi: 10.3389/frmbi.2024.1366948. Epub 2024 Jun 2. PMID: 40687607; PMCID: PMC12276884.


[2] Friedman J., Hastie T., Tibshirani R. (2019)  Graphical Lasso: Estimation of Gaussian Graphical Models, Version: 1.11. https://CRAN.R-project.org/package=glasso. 

